src/utils/json-data.js

```diff
@@ -0,0 +1,135 @@
+const fs = require('fs-extra');
+const path = require('path');
+const reserved = require('reserved-words');
+const shared = require('./shared');
+const glob = require('glob');
+
+// Matches {{ stache.jsonData. }} with any number of spaces between the '{{' and 'stache.',
+// any keys following the 'jsonData.', and the closing '}}'
+const angularBindingRegExp = new RegExp(/\{\{\s*stache.jsonData.*?\}\}/g);
+const buildTimeBindingRegExp = new RegExp(/\{\{\s*@buildtime:\s*stache.jsonData.*?\}\}/g);
+
+let _globalData;
+
+const getGlobalData = () => {
+  if (!_globalData) {
+    _globalData = buildGlobalDataFromJson();
+  }
+
+  return _globalData;
+};
+
+const buildGlobalDataFromJson = () => {
+  const root = shared.resolveAssetsPath('data');
+  const filePaths = glob.sync(path.join(root, '*.json'));
+
+  if (!filePaths.length) {
+    return;
+  }
+
+  const dataObject = filePaths.reduce((acc, filePath) => {
+    const fileName = path.basename(filePath);
+    const propertyName = convertFileNameToObjectPropertyName(fileName);
+
+    if (!isPropertyNameValid(propertyName)) {
+      console.error(
+        new shared.StachePluginError(
+          `A valid Object property could not be determined from file ${fileName}!
+          The property key '${propertyName}' cannot be used. Please choose another file name.`
+        )
+      );
+      return acc;
+    }
+
+    let contents;
+
+    try {
+      contents = fs.readFileSync(filePath);
+    } catch (error) {
+      throw new shared.StachePluginError(error.message);
+    }
+
+    acc[propertyName] = JSON.parse(contents);
+
+    return acc;
+  }, {});
+
+  return dataObject;
+};
+
+const convertFileNameToObjectPropertyName = (fileName) => {
+  return fileName.split('.')[0]
+    .replace(/\s+/g, '_')     // Replace spaces with underscores
+    .replace(/[^\w-]+/g, '')  // Remove all non-word chars
+    .replace(/-/g, '_')       // Replace all dashes with underscores
+    .replace(/--+/g, '_')     // Replace multiple dashes with single underscore
+    .replace(/^-+/, '')       // Trim - from start of text
+    .replace(/-+$/, '');      // Trim - from end of text;
+};
+
+const isPropertyNameValid = (propertyName) => {
+  if (!propertyName || propertyName === 'prototype') {
+    return false;
+  }
+
+  // Parsing as boolean because reserved-words returns `undefined` for falsy values.
+  return !reserved.check(propertyName, 'es6', true);
+};
+
+const parseAllBuildTimeBindings = (content) => {
+  const buildTimeBindings = content.match(buildTimeBindingRegExp);
+
+  if (!buildTimeBindings) {
+    return content;
+  }
+
+  buildTimeBindings.forEach(buildTimeBinding => {
+    let dataValue = parseAngularBindings(buildTimeBinding);
+    content = content.replace(buildTimeBinding, dataValue);
+  });
+
+  return content;
+}
+
+const parseAngularBindings = (unparsedValue) => {
+  let angularBinding = unparsedValue.replace(/@buildtime:/g, '');
```

Comment:
I was working on pulling this out into it's own constant, however the problem comes when building a regex string with escape characters.   I couldn't get it to play friendly with a string being passed into a new regexp.

the value is `@buildtime:`  though all our methods and variables in the javascript code will be the camel case `buildTime`.   This is because I don't want our users to have to worry about that variable being camel case and it looks ugly on the front end.